async/cohttp_async.ml

```diff
@@ -156,25 +156,60 @@ end
 
 module Client = struct
 
+  let read_request ic =
+    Response.read ic >>| function
+    | `Eof -> failwith "Connection closed by remote host"
+    | `Invalid reason -> failwith reason
```

Comment:
Yes @trevorsummerssmith I think that's a better return type. However the connection will probably be terminated after the first error. If the body is corrupted or unreadable somehow it probably doesn't make much sense to attempt reading the next body. So you will get at most one error.

Just making sure that's clear.